File: Sorting Algorithms/Bubble Sort/BubbleSort.c
#include <stdio.h>

void bubbleSort(int arr[], int n) 
{ 
    int i, j, temp; 
    int swapped;
    for (i = 0; i < n-1; i++) 
    { 
        swapped = 0; 
        for (j = 0; j < n-i-1; j++) 
        { 
            if (arr[j] > arr[j+1]) 
            {
                temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                swapped = 1; 
            } 
        } 

    if (swapped == 0) 
        break;
   }
}

void main()
{ 
    int arr[] = {64, 34, 25, 12, 22, 11, 90}; 
    int n = sizeof(arr)/sizeof(arr[0]);

    printf("Array before sort: \n"); 
    for (int i=0; i < n; i++) 
        printf("%d ", arr[i]);
    printf("\n\n");

    bubbleSort(arr, n);

    printf("Sorted array: \n"); 
    for (int i=0; i < n; i++) 
        printf("%d ", arr[i]); 
    printf("\n");
} 

File: Sorting Algorithms/Bubble Sort/BubbleSort.js
// Javascript program for implementation of Bubble Sort
const bubbleSort = (arr) => {
    for (let i = 0; i < arr.length; i++) {
        for (let j = 0; j < arr.length - i - 1; j++) {
            if (arr[j + 1] < arr[j]) {
                // swapping the elements
                [arr[j + 1], arr[j]] = [arr[j], arr[j + 1]];
            }
        }
    }
    return(arr);
}

// driver program
const sortedArray = bubbleSort([5,10,1,3,2,7,8,9,6,4]);
console.log(sortedArray);